diff -rupN OpenCV-2.4.2/modules/flann/include/opencv2/flann/lsh_index.h OpenCV-2.4.2-lsh/modules/flann/include/opencv2/flann/lsh_index.h
--- OpenCV-2.4.2/modules/flann/include/opencv2/flann/lsh_index.h	2012-06-29 10:56:01.000000000 +1200
+++ OpenCV-2.4.2-lsh/modules/flann/include/opencv2/flann/lsh_index.h	2012-09-19 19:13:25.000000000 +1200
@@ -90,9 +90,9 @@ public:
              Distance d = Distance()) :
         dataset_(input_data), index_params_(params), distance_(d)
     {
-        table_number_ = get_param<int>(index_params_,"table_number",12);
-        key_size_ = get_param<int>(index_params_,"key_size",20);
-        multi_probe_level_ = get_param<int>(index_params_,"multi_probe_level",2);
+        table_number_ = get_param<unsigned int>(index_params_,"table_number",12);
+        key_size_ = get_param<unsigned int>(index_params_,"key_size",20);
+        multi_probe_level_ = get_param<unsigned int>(index_params_,"multi_probe_level",2);
 
         feature_size_ = (unsigned)dataset_.cols;
         fill_xor_mask(0, key_size_, multi_probe_level_, xor_masks_);
